### DISCUSSION TITLE
Help on Trello custom provider

### DISCUSSION THREAD

**Discussion by NayanChauveau** [nextauthjs/next-auth#8743] [Help]
### Summary

Hi everyone,

I am trying to create a custom provider for Trello. For trello I have to use the oauth 1.0A at most.

Here is my implementation:

```export const options: NextAuthOptions = {
  providers: [
    {
      id: "trello",
      name: "Trello",
      type: "oauth",
      version: "1.0A",
      accessTokenUrl: "https://trello.com/1/OAuthGetAccessToken",
      requestTokenUrl: "https://trello.com/1/OAuthGetRequestToken",
      authorization: "https://trello.com/1/OAuthAuthorizeToken",
      profileUrl: "https://api.trello.com/1/members/me/",
      profile: (profile) => {
        return {
          id: profile.id,
          name: profile.fullName,
          email: profile.email,
          image: profile.avatarUrl + "/170.png",
        };
      },
      clientId: process.env.TRELLO_CLIENT_ID,
      clientSecret: process.env.TRELLO_CLIENT_SECRET,
    },
  ],
};
```

Everything is working but I want to change the scope of privileges asked by the application. 
This is done by adding parameters to the authorization url: scope=read,write,account

When I use it like it is currently, it is creating a authorization url with the current authorization string + some url parameters. Like this:
`https://trello.com/1/OAuthAuthorizeToken?oauth_token=XXX&oauth_token_secret=XXX&oauth_callback_confirmed=true`

I just need to add the parameter scope to it. 
If I do it by hand, updating this string and adding &scope=read,write,account it works.

But I don't find a solution to make the custom provider do it automatically.

Things I have tried:

Changing the authorization line by:

```
authorization: {
    url: "https://trello.com/1/OAuthAuthorizeToken",
    params: { 
        scope: "read,write,account" 
    },
},
```
      
It seams to work only on oauth 2.

Changing the authorization line by:

```
authorization: {
    url: "https://trello.com/1/OAuthAuthorizeToken?scope=read,write,account",
},
```

It is not working, the others params are added with a question mark like that:

`https://trello.com/1/OAuthAuthorizeToken?scope=read,write,account?oauth_token=XXX&oauth_token_secret=XXX&oauth_callback_confirmed=true`

Does anyone have a solution for this ?
Thanks a lot for your help.

**Comment by gregorym** [comment]
@NayanChauveau Did you ever found a solution?
